psetv2/input.go

```diff
@@ -1034,6 +1118,38 @@ func (i *Input) deserialize(buf *bytes.Buffer) error {
 						return ErrInDuplicatedField("issuance blind inflation keys proof")
 					}
 					i.IssuanceBlindInflationKeysProof = kp.Value
+				case InputExplicitValue:
+					if i.ExplicitValue != 0 {
+						return ErrInDuplicatedField("explicit value")
+					}
+					if len(kp.Value) != 8 {
+						return ErrInInvalidExplicitValue
+					}
+					i.ExplicitValue = binary.LittleEndian.Uint64(kp.Value)
+				case InputValueProof:
+					if len(i.ValueProof) > 0 {
+						return ErrInDuplicatedField("value proof")
+					}
+					i.ValueProof = kp.Value
+				case InputExplicitAsset:
+					if len(i.ExplicitAsset) > 0 {
+						return ErrInDuplicatedField("explicit asset")
+					}
+					if len(kp.Value) != 32 {
+						return ErrInInvalidExplicitAsset
+					}
+				case InputAssetProof:
+					if len(i.AssetProof) > 0 {
+						return ErrInDuplicatedField("asset proof")
+					}
+					i.AssetProof = kp.Value
+				case InputBlindedIssuanceValue:
+					if len(kp.Value) != 1 {
+						return ErrInInvalidBlindedIssuanceValue
+					}
```

Comment:
Missing check for duplicated field:
```suggestion
					if i.BlindedIssuance != nil {
					        return ErrInDuplicatedField("blinded issuance flag")
					}
					
					if len(kp.Value) != 1 {
						return ErrInInvalidBlindedIssuanceValue
					}
```

Reply:
Done